### ISSUE TITLE
[Bug] Missing dependency on `json-schema`

### ISSUE THREAD

**Issue by arcanis** [typescript-eslint/typescript-eslint#2013]
The `json-schema.js` file from the build directory (cf source [here](https://cdn.jsdelivr.net/npm/@typescript-eslint/experimental-utils@2.32.0/dist/json-schema.js)) references the `json-schema` dependency, despite it not being listed in the [manifest](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/experimental-utils/package.json) (`@types/json-schema` is listed, but it doesn't contribute to the actual runtime resolution).

**Repro**

```bash
cd $(mktemp -d)
yarn init -2
yarn add eslint typescript @typescript-eslint/parser @typescript-eslint/eslint-plugin
echo '{"parser": "@typescript-eslint/parser", "plugins": ["@typescript-eslint"], "rules": {"@typescript-eslint/explicit-function-return-type": "error"}}' > .eslintrc
echo 'const f = (): number => 42;' | tee ok.ts
yarn eslint ok.ts
echo 'const f = () => 42;' | tee ko.ts
! yarn eslint ko.ts
```

**Expected Result**

Everything should pass

**Actual Result**

```
Error: Failed to load parser '@typescript-eslint/parser' declared in '.eslintrc': @typescript-eslint/experimental-utils tried to access json-schema, but it isn't declared in its dependencies; this makes the require call ambiguous and unsound.

Required package: json-schema (via "json-schema")
Required by: @typescript-eslint/experimental-utils@virtual:6fc75bf5c6c398c8fde3401485e0f965a9b6ca80d5d9b25914df8ba08830643d0cec44ce00a62cfa15c86d0ff4a84e09e9559f4ab6e85567f6245324482d38f4#npm:2.32.0 (via /tmp/tmp.DSNzJ4IG7b/.yarn/$$virtual/@typescript-eslint-experimental-utils-virtual-7389a78a1b/0/cache/@typescript-eslint-experimental-utils-npm-2.32.0-e9b6e39928-6ea45e3443.zip/node_modules/@typescript-eslint/experimental-utils/dist/)

Require stack:
- /tmp/tmp.DSNzJ4IG7b/.yarn/$$virtual/@typescript-eslint-experimental-utils-virtual-7389a78a1b/0/cache/@typescript-eslint-experimental-utils-npm-2.32.0-e9b6e39928-6ea45e3443.zip/node_modules/@typescript-eslint/experimental-utils/dist/json-schema.js
```

https://github.com/yarnpkg/berry/runs/667366222?check_suite_focus=true#step:6:40

**Additional Info**

**Versions**

| package                                 | version |
| --------------------------------------- | ------- |
| `@typescript-eslint/experimental-utils` | `2.32.0` |

**Comment by bradzacher** [comment]
See pinned issue #2009

**Comment by bradzacher** [comment]
This has now been released to latest - 2.33.0
